website/source/api/agent/service.html.md

```diff
@@ -98,7 +98,9 @@ The table below shows this endpoint's support for
 
 - `Check` `(Check: nil)` - Specifies a check. Please see the
   [check documentation](/api/agent/check.html) for more information about the
-  accepted fields.
+  accepted fields. If you don't provide a name or id for the check then they
+  will be generated. To provide a custom id and/or name set the `CheckID`
+  and/or `Name` field.
```

Comment:
Even though there's deterministic behavior I think we should recommend that you either supply all the IDs or non of them. It's pretty subtle how to do it otherwise.